cfgov/cfgov/urls.py

```diff
@@ -617,5 +617,13 @@ def handle_error(code, request, exception=None):
                             "HTTP Error %s." % str(code), status=code)
 
 
-handler404 = partial(handle_error, 404)
+# Handle case-insensitive URLs
+# Using (?i) in url() patterns is deprecated in Django 2.1
+def handle_404_error(code, request, exception=None):
+    if request.path != request.path.lower():
+        return redirect(request.path.lower())
```

Comment:
Per @kelleyholden's suggestion in GHE/CFGOV/platform/issues/3748#issuecomment-249062, these should be 301s:

```suggestion
        return redirect(request.path.lower(), permanent=True)
```

Reply:
Django comes with an optional redirects application. It lets you store redirects in a database and handles the redirecting for you. It uses the HTTP response status code 301 Moved Permanently by default.

Reply:
@cwdavies the Django redirects application does the same thing our Wagtail redirect application does. But that's not really relevant here.

We need to make sure we're sending permanent, 301 redirects for case corrections. By default [Django's `redirect()` function has `permanent=False`](https://docs.djangoproject.com/en/2.2/topics/http/shortcuts/#redirect), which is a 302. By passing `permanent=True`, it will be a 301.